warehouse/admin/views/classifiers.py

```diff
@@ -92,14 +96,58 @@ def add_child_classifier(self):
     uses_session=True,
     require_methods=False,
     require_csrf=True,
+    request_param=["classifier_id"],
 )
 def deprecate_classifier(request):
-    classifier = request.db.query(Classifier).get(request.params.get("classifier_id"))
-
-    classifier.deprecated = True
+    deprecated_classifier_id = int(request.params.get("classifier_id"))
+    alternative_classifier_ids = {
+        int(alternative_classifier_id)
+        for alternative_classifier_id in request.params.getall("deprecated_by")
+        if alternative_classifier_id
+    }
+
+    deprecated_classifier = request.db.query(Classifier).get(deprecated_classifier_id)
+
+    try:
+        if deprecated_classifier_id in alternative_classifier_ids:
+            raise ValidationException(
+                f"You can not deprecate the classifier in favor of itself"
```

Comment:
Let's just remove the classifier from the list of possible alternative classifiers, and don't worry about explicitly checking it here. Because this UI is just use by the administrators (and relatively infrequently, at that) we don't need to worry too much about these edge cases, I'd rather just simplify the logic as much as possible.